### PR TITLE
Windows discovery [4413]

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request adds discovery functionality for Windows.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -179,6 +181,7 @@ set(SRCS
     $<$<BOOL:${PROFILE_READ_ACCESS}>:src/c/profile/session/read_access.c>
     $<$<BOOL:${PROFILE_WRITE_ACCESS}>:src/c/profile/session/write_access.c>
     $<$<AND:$<NOT:$<PLATFORM_ID:Windows>>,$<BOOL:${PROFILE_DISCOVERY}>>:src/c/profile/discovery/discovery.c>
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply CMakeLists.txt]: Yes, sir!

File: CMakeLists.txt [hunk 2]
```diff
@@ -263,9 +266,9 @@ if(EPROSIMA_BUILD_EXAMPLES)
     add_subdirectory(examples/SubscribeHelloWorld)
     add_subdirectory(examples/ShapesDemo)
     add_subdirectory(examples/Deployment)
-    if(PLATFORM_NAME_LINUX OR PLATFORM_NAME_NUTTX)
+#    if(PLATFORM_NAME_LINUX OR PLATFORM_NAME_NUTTX)
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply CMakeLists.txt]: Yes, it was a mistake. Removed.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/117/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/78/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/59/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/118/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/79/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/60/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/123/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/84/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/65/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Linux/127/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20Windows/88/)
* NuttX [Build Status](http://jenkins.eprosima.com:8080/job/Micro%20XRCE-DDS%20Client%20Manual%20NuttX/69/)

**Comment by lemunozm** [comment]
There is a redmine related task to enable the integration test in windows for discovery :4541

**Pull request merged**